### ISSUE TITLE
CohortCharacterizationServiceTests fail

### ISSUE THREAD

**Issue by chrisknoll** [OHDSI/WebAPI#1506]
The CohortCharacterizationServiceTests: testExportGeneration() and testExportGenerationWithStrata() are failing.

I found 2 problems:

1) When reading in the characterization's feature analysis design, one of the checks is to determine if the "aggregate" node exists:

```
                if (CcResultType.DISTRIBUTION == dto.getStatType()) {
                    JsonNode aggregate = node.get("aggregate");
                    if (!aggregate.isNull()) {
                        FeAnalysisAggregateDTO aggregateValue = objectMapper.readValue(aggregate.traverse(), FeAnalysisAggregateDTO.class);
                        ((FeAnalysisWithConceptSetDTO)dto).setAggregate(aggregateValue);
                    }
                }

```

In the test, the aggregate is null, so a simple fix is to make IF: if (aggregate != null && !aggregate.isNull().

2) After making the fix for 1) locally, I was able to get the entire test to execute, but it fails.  To be honest, this test is very hard to follow:  It takes a CC design to generate (no information on what is in the design) and then after generation completes, it uses a call to CcServiceImpl.findData() to extract specific elements of the results from the generation into a zip (also, no details on which elements it is trying to extract is given in the test description).  The problem:  The design has only the following FE Analysis IDs defined: `133,118,134,135,121`.  When `findResult()` is called, the params indicates only analysisIds: `[105]`.  It is not clear where this is comming from, except the call to `checkRequest` is from the `PARAM_JSON` defined in the test class. For some reason, the analyses in the CC design to not match the export paramaters.  It's hard to tell, but I believe there's even cohort definition id references in the params that aren't in the CC design.   Due to that these element do not exist when it tries to export the .zip, it throws an exception.

**Comment by chrisknoll** [comment]
Ok, Update on 1:

Looks like I was working off of a PR branch that wasn't up to date on master, so the whole branch of code related to `aggregate.isNull` is gone from the codebase, so that's not a problem.

The second problem, tho, remains:  One of the changes in master was to throw an exception if an invalid analysisID was requested in the param filter.  Some of the test cases are checking for an 'empty' result if you pass in invalid analysisId, but as of December 2019, that's no longer valid. So, those test cases will be either changed or we will allow invalid analysisIDs.

Another issue is that the PRESET analysisIDs may not be matching up with the inputs in the test case, so we'll investigate that.

I've asked @azimov to take a look at this, so I will assign this issue over to him for now.

**Comment by chrisknoll** [comment]
Hi, @olga-ganina , @wivern , @ssuvorov-fls ,
  When reviewing this test, there seems to be about 80 sub-tests defined in the `reportData.json` which we're trying to reconcile with the current behavior of the application...this means renumbering some of the `analysis_id`s in the file and also evaluating the resulting filenames/file sizes that are returned.  

@azimov reports:

> Basically what happened was it was ignoring the default features but they should be there as they’re specified in the design. So that means the files are missing and the file lengths are wrong 

The question is, are we interpreting the behavior of the code properly or is the test defining the proper behavior and the code is wrong?  We'd like to get your input on this behavior so that we can adjust the test or the code properly.

**Comment by azimov** [comment]
So, currently the analysisID's set in reportData.json and reportDataStratified.json are off by 1. I'm not sure why this is the case but it would be good to know how these test param files were generated; I'm sure this wasn't by hand.

The code has also been updated since the test was written, namely the reports for any analysis ids specified in the design are exported when no Ids are set.

The file lengths in reportDataStratified.json are also wrong, I'm not sure why but there is nothing wrong with the exported reports (as far as I can tell) they all appear to have the correct ids. The branch I'm working on will show what I mean by this. Again I, can't really tell why this is because I'm not sure how the test json files were generated.

**Comment by chrisknoll** [comment]
Let's get some feedback from the original devs (if they are available) in the PR, and then if no one has any objections, we'll proceed with your proposed fixes.
